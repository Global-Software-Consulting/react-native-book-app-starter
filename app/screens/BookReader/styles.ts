import { StyleSheet } from 'react-native';
import { getPercentageHeight, getPercentageWidth } from 'utils/dimentionUtil';
export const useStyles = () => {
    const height = getPercentageHeight();
    const width = getPercentageWidth();


    const styles = StyleSheet.create({
        mainView:{
            flex: 1,
            justifyContent: 'flex-start',
            alignItems: 'center',
        },
        pdfView:
        { height:height('100%') , width: width('100%') }
        
    });
    return styles;
};
