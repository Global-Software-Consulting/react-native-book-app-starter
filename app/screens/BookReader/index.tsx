import React from 'react'
import { View } from 'react-native'
<<<<<<< HEAD
// import Pdf from 'react-native-pdf';
=======
import Pdf from 'react-native-pdf'
>>>>>>> 44ebcc3db372a8f9d63d4872f4fe55409e5c797f
import { useStyles } from './styles'
const BookReader: React.FC = () => {
    const src = {
        uri: 'https://www.adelaide.edu.au/writingcentre/sites/default/files/docs/learningguide-articlesinenglishgrammar.pdf',
    }
    const styles = useStyles()

<<<<<<< HEAD
    return <View style={styles.mainView}>{/* <Pdf source={src} style={styles.pdfView} /> */}</View>
=======
    return (
        <View style={styles.mainView}>
            <Pdf source={src} style={styles.pdfView} />
        </View>
    )
>>>>>>> 44ebcc3db372a8f9d63d4872f4fe55409e5c797f
}
export default BookReader
