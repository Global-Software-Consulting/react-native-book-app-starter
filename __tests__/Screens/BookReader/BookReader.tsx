import { render } from '@testing-library/react-native'
import React from 'react'
import BookReader from 'screens/BookReader'

describe('Component testing', () => {
    test('Snapshot ', async () => {
        const tree = render(<BookReader />)
        expect(tree).toMatchSnapshot()
    })
})
