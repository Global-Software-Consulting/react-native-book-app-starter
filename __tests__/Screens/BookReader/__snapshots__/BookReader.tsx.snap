// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing Snapshot  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
<<<<<<< HEAD
/>
=======
>
  <Pdf
    source={
      Object {
        "uri": "https://www.adelaide.edu.au/writingcentre/sites/default/files/docs/learningguide-articlesinenglishgrammar.pdf",
      }
    }
    style={
      Object {
        "height": 1334,
        "width": 750,
      }
    }
  />
</View>
>>>>>>> 44ebcc3db372a8f9d63d4872f4fe55409e5c797f
`;
